### PR TITLE
TASK-6347 Fix normalization issue in ClinVar

### PR DESCRIPTION
**Pull request by jtarraga**
[TASK-6347](https://app.clickup.com/t/8694qkzv5) normalization issue in ClinVar

### CODE REVIEW

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/ClinVarIndexer.java [hunk 1]
```diff
@@ -266,13 +266,34 @@ private boolean updateRocksDB(AlleleLocationData alleleLocationData, PublicSetTy
...
-                        .getClinicalSignificance()
-                        .getDescription();
-                String reviewStatusName = publicSet.getReferenceClinVarAssertion().getClinicalSignificance()
-                        .getReviewStatus().name();
-                List<ObservationSet> getObservedIn = publicSet.getReferenceClinVarAssertion().getObservedIn();
+                String accession = null;
+                try {
+                    accession = publicSet.getReferenceClinVarAssertion().getClinVarAccession().getAcc();
+                } catch (Exception e) {
+                    logger.warn("Error getting accession\n" + StringUtils.join(e.getStackTrace(), "\n"));
```
Comment (j-coll) [inline cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/ClinVarIndexer.java:273]: Better let the logger to print the stacktrace:
```
logger.warn("Error getting accession", e);
```

and you can also indicate what will happen with the accession:
```
logger.warn("Error getting accession. Ignore error and leave accession as null.", e);
```
This applies to all other fields read in this method

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java [hunk 2]
```diff
@@ -56,36 +59,31 @@ public GwasIndexer(Path gwasFile, Path dbSnpTabixFile, Path genomeSequenceFilePa
...
 
             long processedGwasLines = 0;
 
-            logger.info("Parsing GWAS catalog file ...");
+            logger.info("Parsing GWAS catalog file {} ...", gwasFile);
+            String line;
             while ((line = inputReader.readLine()) != null) {
+                ++lineCounter;
                 if (!line.isEmpty()) {
                     processedGwasLines++;
```
Comment (j-coll) [inline cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:79]: Might want to use a `org.opencb.commons.ProgressLogger` here

```java
ProgressLogger progressLogger = new ProgressLogger("Lines parsed").setBatchSize(10000);

...
if (!line.isEmpty() ) {
  progressLogger.increment(1);
}
```

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java [hunk 3]
```diff
@@ -56,36 +59,31 @@ public GwasIndexer(Path gwasFile, Path dbSnpTabixFile, Path genomeSequenceFilePa
...
     public void index() throws RocksDBException, IOException {
-        logger.info("Parsing GWAS catalog file ...");
-
-        BufferedReader inputReader = null;
-        TabixReader dbsnpTabixReader = null;
-
-        try {
-            logger.info("Opening GWAS catalog file " + gwasFile + " ...");
-            inputReader = new BufferedReader(new FileReader(gwasFile.toFile()));
+        try (BufferedReader inputReader = new BufferedReader(new FileReader(gwasFile.toFile()));
```
Comment (j-coll) [inline cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:62]: Beautiful!

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java [hunk 4]
```diff
@@ -270,21 +262,27 @@ private void processGwasCatalogLine(String[] values, TabixReader dbsnpTabixReade
...
                     }
-                    try {
-                        scores.setPValueMlog(Double.parseDouble(values[28]));
-                    } catch (NumberFormatException e) {
-//                        logger.warn(e.getMessage() + ". Parsing pValue mlog: " + values[28]);
+                    if (StringUtils.isNotEmpty(values[28])) {
+                        try {
+                            scores.setPValueMlog(Double.parseDouble(values[28]));
+                        } catch (NumberFormatException e) {
+                            logger.warn(e.getMessage() + ". Parsing pValue mlog: " + values[28]);
```
Comment (j-coll) [inline cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:276]: I understand that this exception will happen be too ofter to print the whole stacktrace, which would not add much information

**Review by j-coll — CHANGES_REQUESTED** [review]

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java (hunk 2 above)
Comment (j-coll) [reply cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:79]: This is just a comment, no action is required (i.e. it's optional)

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java (hunk 4 above)
Comment (jtarraga) [reply cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:276]: Exact.

File: cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java (hunk 2 above)
Comment (jtarraga) [reply cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:79]: Hmm, that's useful, but I can't find a method to get the current number of parsed lines. I need this information to include in the log messages when parsing fails.

Comment (j-coll) [reply cellbase-lib/src/main/java/org/opencb/cellbase/lib/builders/clinical/variant/GwasIndexer.java:79]: That is very inconvenient... there is no way of obtaining that..

**Review by j-coll — APPROVED** [review]

**Pull request merged**